qa/vagrant-helpers.rb

```diff
@@ -34,10 +50,17 @@ def self.parse(lines)
 
     def self.execute(cmd)
       Open3.popen3(cmd) do |stdin, stdout, stderr, wait_thr|
-        { :stdout => stdout.read.chomp, :stderr => stderr.read.chomp,
-          :exit_status => wait_thr.value.exitstatus }
+        ExecuteResponse.new(stdin, stdout.read.chomp, stderr.read.chomp, wait_thr.value.exitstatus)
       end
     end
 
+    def self.execute_successfully(cmd)
```

Comment:
I don't agree with that, If I want to reuse the executor I want to control the behavior of it to make specific assertions. To be clear I want to reuse that code for the CLI test and make custom helper that will handle the right feedback and not depend on the implementation exception.

Its pretty common to use a `!` in ruby if the method can raise an exception. See `save and save!` in ActiveRecord.